src/test/java/org/folio/search/support/base/BaseIntegrationTest.java

```diff
@@ -0,0 +1,86 @@
+package org.folio.search.support.base;
+
+import static java.lang.String.format;
+import static org.folio.dbschema.ObjectMapperTool.getMapper;
+import static org.folio.search.sample.SampleInstances.getSemanticWeb;
+import static org.folio.search.utils.TestUtils.asJsonString;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import org.folio.okapi.common.XOkapiHeaders;
+import org.folio.search.model.ResourceEventBody;
+import org.folio.search.model.rest.request.IndexRequestBody;
+import org.junit.jupiter.api.BeforeAll;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.springframework.test.web.servlet.MockMvc;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Testcontainers
+public abstract class BaseIntegrationTest {
+  private static final String TENANT = "test";
+  private static final String INSTANCE_RESOURCE = "instance";
+
+  @Autowired
+  protected MockMvc mockMvc;
+
+  @Container
+  private final static GenericContainer<?> esContainer = new GenericContainer<>(
+    new ImageFromDockerfile("test-container-embedded-es:7.10.1", false)
+      .withDockerfile(Path.of("docker/elasticsearch/Dockerfile")))
+    .withEnv("discovery.type", "single-node")
+    .withExposedPorts(9200);
```

Comment:
I am not sure about jenkins, but I guess it would be better if port will be random. It would fail if I have elasticsearch running locally at 9200 and I want to run unit tests to check functionality and push some changes